### PR TITLE
LPS-185224 Enable editing of newly added FDS Fields

### PR DESCRIPTION
**Pull request by markocikos**
### References

- [LPS-185224 Enable editing of newly added FDS Fields](https://issues.liferay.com/browse/LPS-185224)

### What is the goal of this PR?

We are enabling editing of newly added item, by fixing two bugs:
- We need to return ERC of newly added item, because we are now using ERC endpoint
- `id`, a `Long`, is serialized as a string, so we need to transform it to a number before we can compare it.

### What does it look like?

https://github.com/liferay-frontend/liferay-portal/assets/5435511/47a3fe7a-b6ea-4cb9-8151-c126479174d8

### Steps to reproduce

1. Add `feature.flag.LPS-164563=true` to `portal-ext.properties`. This enables FDS Views Manager.
1. Add `feature.flag.LPS-161364=true` to `portal-ext.properties`. This enables Objects relationships nested fields, used by FDS Views Manager.
1. If you previously opened Datasets page, delete `FDS Entry` and `FDS View` on Global Menu > Control Panel > Object > Objects
1. Open Global Menu > Control Panel > Object > Datasets
1. Create a dataset, and a view for that dataset. Edit view.

### CODE REVIEW

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Fields.tsx [hunk 1]
```diff
@@ -145,7 +145,10 @@ const SaveFDSFieldsModalContent = ({
 		});
 
 		onSave({
-			createdFDSFields,
+			createdFDSFields: createdFDSFields.map((fdsField) => ({
+				...fdsField,
+				id: Number(fdsField.id),
```
Comment (bryceosterhaus) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Fields.tsx:150]: How come you had to change this here? Is `id` not a number already?

**Review by bryceosterhaus — APPROVED** [review]
LGTM, just one question but not a blocker

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Fields.tsx (hunk 1 above)
Comment (markocikos) [reply modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Fields.tsx:150]: It's not for the JSON returned by our save action. The root cause is that we are [serializing `long` as a string](https://github.com/liferay/liferay-portal/blob/master/portal-impl/src/com/liferay/portal/json/JSONObjectImpl.java#L305), and [`id` is a long](https://github.com/liferay/liferay-portal/blob/master/modules/apps/object/object-api/src/main/java/com/liferay/object/model/ObjectEntryModel.java#L117).

**Comment by markocikos** [comment]
ci:forward

**Pull request closed without merging**